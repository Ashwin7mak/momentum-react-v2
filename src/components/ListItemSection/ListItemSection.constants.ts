const CLASS_PREFIX = 'md-list-item-section';

const DEFAULTS = {};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { DEFAULTS, STYLE };
