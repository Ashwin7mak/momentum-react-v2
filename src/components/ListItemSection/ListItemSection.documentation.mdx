`<ListItemSection />` component can be used inside `<ListItem />` to wrap content and display it in a certain position.
