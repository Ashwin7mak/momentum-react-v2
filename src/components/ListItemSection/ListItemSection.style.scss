.md-list-item-section-wrapper {}
