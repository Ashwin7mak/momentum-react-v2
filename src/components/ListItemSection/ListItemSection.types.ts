import { ReactNode } from 'react';

export interface Props {
  /**
   * className prop description
   */
  className?: string;
  children?: ReactNode;
  position?: 'start' | 'middle' | 'end' | 'fill';
}
