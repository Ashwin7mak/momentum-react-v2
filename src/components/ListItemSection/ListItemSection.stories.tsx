import React, { FC } from 'react';
import { Story } from '@storybook/react';

import ListItemSection, { ListItemSectionProps } from './';
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  PRIMARY_STORY,
} from '@storybook/addon-docs';

import Documentation from './ListItemSection.documentation.mdx';
import ListItem from '../ListItem';
import argTypes from './ListItemSection.stories.args';

const DocsPage: FC = () => (
  <>
    <Title />
    <Subtitle />
    <Description />
    <Documentation />
    <Primary />
    <ArgsTable story={PRIMARY_STORY} />
  </>
);

export default {
  title: 'Momentum UI/ListItemSection',
  component: ListItemSection,
  parameters: {
    expanded: true,
    docs: {
      page: DocsPage,
    },
  },
  argTypes: {
    ...argTypes,
  },
};

const MultiTemplate: Story<ListItemSectionProps> = (args: ListItemSectionProps, { parameters }) => {
  const { variants } = parameters;

  const items = variants.map((variant, index: number) => (
    <div key={index}>
      <ListItem style={{ minWidth: '20rem', border: '1px solid red' }}>
        <ListItemSection {...args} {...variant} style={{ border: '1px solid blue' }} />
      </ListItem>
      <p>{variant.label}</p>
    </div>
  ));

  return (
    <div
      style={{
        display: 'grid',
        gridTemplateColumns: `repeat(1, auto)`,
        gap: '1.5rem',
        alignItems: 'end',
      }}
    >
      {items}
    </div>
  );
};

const Positions = MultiTemplate.bind({});

Positions.args = {
  children: 'Content',
};

Positions.parameters = {
  variants: [
    { position: 'start', label: 'Start' },
    { position: 'middle', label: 'Middle' },
    { position: 'end', label: 'End' },
    { position: 'fill', label: 'Fill' },
  ],
};

export { Positions };
