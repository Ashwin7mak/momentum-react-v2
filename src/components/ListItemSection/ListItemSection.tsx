import React, { FC } from 'react';

import './ListItemSection.style.scss';
import { Props } from './ListItemSection.types';

const ListItemSection: FC<Props> = (props: Props) => {
  const { className, children, position, style, id } = props;

  return (
    <div data-position={position} className={className} style={style} id={id}>
      {children}
    </div>
  );
};

/**
 * Component that can be used inside ListItem to wrap content and display
 * it in a certain position.
 */

export default ListItemSection;
