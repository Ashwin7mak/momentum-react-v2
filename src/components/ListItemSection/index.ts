import { default as ListItemSection } from './ListItemSection';
import { Props } from './ListItemSection.types';
import * as CONSTANTS from './ListItemSection.constants';

export { CONSTANTS as LIST_ITEM_SECTION_CONSTANTS };

export type ListItemSectionProps = Props;

export default ListItemSection;
