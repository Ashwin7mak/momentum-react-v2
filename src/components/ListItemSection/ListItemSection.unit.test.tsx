import ListItemSection from '.';
import { mount } from 'enzyme';
import React from 'react';

describe('ListItemSection', () => {
  let container;

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<ListItemSection>Test</ListItemSection>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      container = mount(<ListItemSection className={className}>Test</ListItemSection>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      container = mount(<ListItemSection id={id}>Test</ListItemSection>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      container = mount(<ListItemSection style={style}>Test</ListItemSection>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with position', () => {
      expect.assertions(1);

      const position = 'end';

      container = mount(<ListItemSection position={position}>Test</ListItemSection>);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      container = mount(<ListItemSection className={className}>Test</ListItemSection>);

      const element = container.find(ListItemSection).getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      container = mount(<ListItemSection id={id}>Test</ListItemSection>);

      const element = container.find(ListItemSection).getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      container = mount(<ListItemSection style={style}>Test</ListItemSection>);

      const element = container.find(ListItemSection).getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided position when style is position', () => {
      expect.assertions(1);

      const position = 'end';

      container = mount(<ListItemSection position={position}>Test</ListItemSection>);

      const element = container.find(ListItemSection).getDOMNode();

      expect(element.getAttribute('data-position')).toBe(position);
    });
  });
});
