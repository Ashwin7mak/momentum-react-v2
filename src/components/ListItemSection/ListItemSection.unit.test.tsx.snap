// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemSection snapshot should match snapshot 1`] = `
<ListItemSection>
  <div>
    Test
  </div>
</ListItemSection>
`;

exports[`ListItemSection snapshot should match snapshot with className 1`] = `
<ListItemSection
  className="example-class"
>
  <div
    className="example-class"
  >
    Test
  </div>
</ListItemSection>
`;

exports[`ListItemSection snapshot should match snapshot with id 1`] = `
<ListItemSection
  id="example-id"
>
  <div
    id="example-id"
  >
    Test
  </div>
</ListItemSection>
`;

exports[`ListItemSection snapshot should match snapshot with position 1`] = `
<ListItemSection
  position="end"
>
  <div
    data-position="end"
  >
    Test
  </div>
</ListItemSection>
`;

exports[`ListItemSection snapshot should match snapshot with style 1`] = `
<ListItemSection
  style={
    Object {
      "color": "pink",
    }
  }
>
  <div
    style={
      Object {
        "color": "pink",
      }
    }
  >
    Test
  </div>
</ListItemSection>
`;
